File: frontend/src/service/leaderboardService.ts
import api from "../config/axios";
import type { ApiResponse } from "../types/ApiResponse";

const API_BASE_URL = "/leaderboards";

export const leaderboardService = {
  getTop100ByCategory: async (
    categoryId: number,
  ): Promise<ApiResponse<any[]>> => {
    const response = await api.get(`${API_BASE_URL}/${categoryId}/top100`);
    return response.data;
  },

  getMyLeaderboardStats: async (
    categoryId: number,
  ): Promise<ApiResponse<any>> => {
    const response = await api.get(`${API_BASE_URL}/${categoryId}/me`);
    return response.data;
  },
};

export default leaderboardService;

File: mobile/lib/services/leaderboard_service.dart
import '../config/api_client.dart';
import '../models/leaderboard_model.dart';

class LeaderboardService {
  final String _baseUrl = '/leaderboards';

  Future<List<LeaderboardEntryResponse>> getTop100ByCategory(
    int categoryId,
  ) async {
    final res = await apiClient.get('$_baseUrl/$categoryId/top100');
    final data = res.data['result'] as List;
    return data.map((json) => LeaderboardEntryResponse.fromJson(json)).toList();
  }

  Future<MyLeaderboardStatsResponse> getMyLeaderboardStats(
    int categoryId,
  ) async {
    final res = await apiClient.get('$_baseUrl/$categoryId/me');
    return MyLeaderboardStatsResponse.fromJson(res.data['result']);
  }
}

final leaderboardService = LeaderboardService();
